### DISCUSSION TITLE
unexpected identifier error after upgrading to babel 7

### DISCUSSION THREAD

**Discussion by jbuhacoff** [babel/babel#12549] [Q&A] [answered]
I'm getting errors from the babel runtime helpers when building my project. I don't know if it's an issue in babel or webpack or if it's a configuration issue, or something else.

My project uses gridsome and vue, which use webpack and babel, and also my project uses commitlint which uses babel.

The errors I'm getting only appear when I do a production build by running `gridsome build`. When I run the development server locally by running `gridsome develop`, everything is working fine.

So during production build I'm getting this error in `@babel/runtime/helpers/esm/unsupportedIterableToArray.js`:

```
import arrayLikeToArray from "@babel/runtime/helpers/esm/arrayLikeToArray";
       ^^^^^^^^^^^^^^^^

SyntaxError: Unexpected identifier
    at Module._compile (internal/modules/cjs/loader.js:720:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at ./node_modules/vue-server-renderer/build.prod.js:1:77645
    at Object.YbeR (webpack:/external "@babel/runtime/helpers/esm/unsupportedIterableToArray":1:0)
    at __webpack_require__ (webpack/bootstrap:25:0)
    at Module.O/VM (assets/js/app.5982aa1e.js:1844:35)
```

If I edit that file and change the import and export lines to commonJS, the build stops with a similar error in another file instead.

So when I edited that file and changed the first line to this:

```
const { arrayLikeToArray } = require("@babel/runtime/helpers/esm/arrayLikeToArray");
```

I got this error:

```
export default function _unsupportedIterableToArray(o, minLen) {
^^^^^^

SyntaxError: Unexpected token export
```

And when I edited the file again to make the second line look like this:

```
module.exports = function _unsupportedIterableToArray(o, minLen) {
```

That file compiled ok and I got another syntax error about the export in another file (`@babel/runtime/helpers/esm/arrayLikeToArray.js`).

Here is the content of babel.config.js:

```
module.exports = {
    presets: [
        '@vue/cli-plugin-babel/preset',
    ],
};
```

Other presets I tried were `@vue/babel-preset-app` and previously `@vue/app`, but changing that doesn't seem to make any difference.

Any ideas?

I have reproduced the issue with a small project that you can see here:  https://github.com/jbuhacoff/website-broken-html5-20201221

**Comment by nicolo-ribaudo** [comment, marked as answer]
It's because `@vue/cli-plugin-babel/preset` tells Babel to inject helpers containing ES modules ([here](https://github.com/vuejs/vue-cli/blob/451f3d2f5fae5d153d337f00e35be9571638101c/packages/%40vue/babel-preset-app/index.js#L252)).
You can try running your build command with the `VUE_CLI_BABEL_TRANSPILE_MODULES=true` env variable.

A very similar problem has also been reported at https://github.com/gridsome/gridsome/issues/1385.

**Comment by jbuhacoff** [comment]
Thanks for that suggestion. I'm continuing the discussion over there.
